packages/store-indexer/src/postgres/getLogs.ts

```diff
@@ -0,0 +1,69 @@
+import { PgDatabase } from "drizzle-orm/pg-core";
+import { Hex } from "viem";
+import { StorageAdapterLog, SyncFilter } from "@latticexyz/store-sync";
+import { tables } from "@latticexyz/store-sync/postgres";
+import { and, eq, or } from "drizzle-orm";
+import { decodeDynamicField } from "@latticexyz/protocol-parser";
+import { bigIntMax } from "@latticexyz/common/utils";
+
+export async function getLogs(
+  database: PgDatabase<any>,
+  {
+    chainId,
+    address,
+    filters = [],
+  }: {
+    readonly chainId: number;
+    readonly address?: Hex;
+    readonly filters?: readonly SyncFilter[];
+  }
+): Promise<{ blockNumber: bigint; logs: (StorageAdapterLog & { eventName: "Store_SetRecord" })[] }> {
+  const conditions = filters.length
+    ? filters.map((filter) =>
+        and(
+          address != null ? eq(tables.recordsTable.address, address) : undefined,
+          eq(tables.recordsTable.tableId, filter.tableId),
+          filter.key0 != null ? eq(tables.recordsTable.key0, filter.key0) : undefined,
+          filter.key1 != null ? eq(tables.recordsTable.key1, filter.key1) : undefined
+        )
+      )
+    : address != null
+    ? [eq(tables.recordsTable.address, address)]
+    : [];
+
+  const chainState = await database
+    .select()
+    .from(tables.chainTable)
+    .where(eq(tables.chainTable.chainId, chainId))
+    .execute()
+    .then((rows) => rows.find(() => true));
+  let blockNumber = chainState?.lastUpdatedBlockNumber ?? 0n;
+
+  const records = await database
+    .select()
+    .from(tables.recordsTable)
+    .where(or(...conditions));
+  blockNumber = bigIntMax(
+    blockNumber,
+    records.reduce((max, record) => bigIntMax(max, record.lastUpdatedBlockNumber ?? 0n), 0n)
+  );
```

Comment:
I only did this here because of the order of queries that are run separately, one for the chain's current block number and one for all the records. I could probably do this at query time (i.e. a single query), but I was being lazy about crafting that query since Drizzle is a bit clunky/annoying with joins.

The issue I am solving for is that the indexer may be up to date for a chain but a world may not have had any recent activity. If we query for just records and return the highest block number of those records, that would signal to the client that it needs to start fetching from the RPC from a much older block number than is necessary.

So basically: we get the chain's block number first, as the "minimum" block number value. Then we get all the records and their highest block number. If it's higher than the chain, then we know some data updated between our two queries and we should use the higher number. If it's lower than the chain, we use the chain block number, because that's where the indexer is at in terms of up-to-date data.